applications/tari_base_node/src/parser.rs

```diff
@@ -1686,16 +1622,72 @@ impl Parser {
-            let command_str =
-                (start_amount.0 + amount_inc.0 * i as u64).to_string() + " " + &dest_pubkey.to_string() + " " + &msg;
-            let args = command_str.split_whitespace();
-            // Execute
-            self.process_send_tari(args);
-        }
+        let mut dht = self.discovery_service.clone();
+        let executor = self.executor.clone();
+        let wallet_transaction_service = self.wallet_transaction_service.clone();
+        self.executor.spawn(async move {
+            // Ensure a valid connection is available by forcing a peer discovery. This is intended to be
+            // a blocking operation before the test starts.
+            match dht
+                .discover_peer(
+                    Box::from(dest_pubkey.clone()),
+                    NodeDestination::PublicKey(Box::from(dest_pubkey.clone())),
+                )
+                .await
+            {
+                Ok(_p) => {
+                    // Wait until specified test start time
+                    let millis_to_wait = (time_utc_start - Utc::now()).num_milliseconds();
+                    println!(
+                        "`make-it-rain` to peer '{}' scheduled to start at {}: msg \"{}\"",
+                        &key, time_utc_start, &msg
+                    );
+                    if millis_to_wait > 0 {
+                        tokio::time::delay_for(Duration::from_millis(millis_to_wait as u64)).await;
+                    }
+
+                    // Send all the transactions
+                    let start = Utc::now();
+                    for i in 0..number_of_txs {
+                        // Manage Tx rate
+                        let millis_actual_i = (Utc::now() - start).num_milliseconds() as u64;
+                        let millis_target_i = (i as f64 / (tx_per_s / 1000.0)) as u64;
+                        if millis_target_i - millis_actual_i > 0 {
+                            // Maximum delay between Txs set to 120 s
+                            tokio::time::delay_for(Duration::from_millis(
+                                (millis_target_i - millis_actual_i).min(120_000u64),
+                            ))
+                            .await;
+                        }
+                        // Send Tx
+                        let wallet_transaction_service = wallet_transaction_service.clone();
+                        let dest_pubkey = dest_pubkey.clone();
+                        let msg = msg.clone();
+                        executor.spawn(async move {
+                            send_tari(
+                                start_amount + amount_inc * (i as u64),
+                                dest_pubkey,
+                                msg,
+                                wallet_transaction_service,
+                            )
+                            .await;
+                        });
```

Comment:
```suggestion
                        executor.spawn(
                            send_tari(
                                start_amount + amount_inc * (i as u64),
                                dest_pubkey,
                                msg,
                                wallet_transaction_service,
                            )
                           
                        );
```

It feels like spawning `n` tasks is redundant because send_transaction already does this internally after preparing the transaction. So the gains in tps should be minimal - that being said, I don't think there's much harm either so let's see.